### PR TITLE
revamp of npm scripts

### PR DESCRIPTION
**Pull request by yuval-klotho**
Rather than hard-coding various npm scripts in the actual package.json, farm them out to a lib and some simple sub-commands scripts.

**[updated]** To use it, do one of

* `npm run klotho build`: Does all the npm, klotho, and pulumi build steps (including `npm install` if needed, and in the compiled dir)
* `npm run pulumi ...`: forwards to `pulumi`, with the appropriate stack name and cwd
    * this also exports `PULUMI_CONFIG_PASSPHRASE=''` _if_ you don't have either `PULUMI_CONFIG_PASSPHRASE` or `PULUMI_CONFIG_PASSPHRASE_FILE` set; Klotho's default generated stack requires an empty password

Note that this sets up a non-standard directory structure for klotho output, but only if you use an app name other than the default (by setting an env var). Instead of `compiled`, it's `compiled-klotho/$app_name`. This is specifically to keep prod and dev compilations/deployments separate, which is an important use case for me. But I'm flexible on it, and especially on the specific dir structure. 

Updated documentation: CloudCompilers/docs#93

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,30 +10,34 @@ To get up and running quickly, see our [installation tutorial][tutorial].
...
 
-```bash
-npm run klotho:deploy
-```
+   ```bash
+   npm run klotho pulumi config set aws:region <YOUR DESIRED REGION> # e.g. us-east-1
+   ```
+2. Make sure you have your env set up with AWS credentials (see [our tutorial][tutorial-aws].
+3. ```bash
+   npm run klotho pulumi up
```
Comment (gordon-klotho) [inline README.md:33]: I dislike `klotho pulumi` - we're not involved in this part and this gives the wrong impression.

Comment (yuval-klotho) [reply README.md:33]: Do you dislike the specific invocation, or the idea in general?

I've found it's pretty handy to have a command that "just works", including (a) finding the right `compiled*/` dir that klotho created, and (b) finding the stack name. With these commands, I just export or unset `KLOTHO_APP_NAME` and I'm good to go.

(I think we are at least a bit involved: we created the dir with the pulumi contents, and seeded the stack.)

Comment (yuval-klotho) [reply README.md:33]: btw, I originally had this as specific commands:

```
npm run klotho up
npm run klotho config <set ...>
npm run klotho down
```

... but all of these were just invoking pulumi with the dir + stack, so I thought it's more straightforward to do that.

Comment (gordon-klotho) [reply README.md:33]: I would prefer it being `npm run pulumi up`. I don't want to give the impression that either a) We = Pulumi b) They can get Pulumi support from us c) That we are replacing/wrapping Pulumi. We're involved in the generation of the files, but that's the boundary. We're "slotting" into a release process, but not at the deployment steps - we're pre-deployment. We can discuss as a group

**Comment by yuval-klotho** [comment]
Updated to have:

* `npm run klotho` does all the building
    * npm install, npm build, npm test, and also the `npm install --prefix compiled`
* `npm run pulumi <...>` runs pulumi with the right dir and stack name

**Review by gordon-klotho — APPROVED** [review]
noice

**Pull request merged**